### PR TITLE
Avoid head-on collisions

### PR DESCRIPTION
**Pull request by JNotelddim**
Sorry it shows all the feature/dev-create-brain branch commits since i branched off that. will squash at the end anyways tho.

### CODE REVIEW

File: src/board.py [hunk 1]
```diff
@@ -13,15 +13,15 @@ def __init__(self, data):
     @property
     def snakes(self) -> List[Snake]:
         """Retreive the list of snakes from the board data."""
-        if not self._snakes:
+        if not self._snakes or len(self._snakes) < 1:
```
Comment (mattcasiro) [inline src/board.py]: Maybe `if self._snakes is not None:`

File: src/board.py [hunk 2]
```diff
@@ -13,15 +13,15 @@ def __init__(self, data):
...
             snakes = [Snake(snake_data) for snake_data in self._data['snakes']]
             self._snakes = snakes
         return self._snakes
 
     @property
     def foods(self) -> List[Coordinate]:
         """Retreive the list of food from the board data."""
-        if not self._foods:
+        if not self._foods or len(self._foods) < 1:
```
Comment (mattcasiro) [inline src/board.py]: Maybe `if self._foods is not None:`

File: src/brain.py [hunk 3]
```diff
@@ -43,3 +44,10 @@ def get_nearest_food(self) -> Optional[Coordinate]:
             return None
 
         return closest_food[0]
+
+    def get_threatening_snakes_moves(self) -> List[Coordinate]:
+        """Get the coordinates which will result in head-on collisions.""" 
+        #note, we can ignore snakes smaller than us.
+
```
Comment (mattcasiro) [inline src/brain.py]: unnecessary whitespace, since there's another change to make anyway

**Review by mattcasiro — CHANGES_REQUESTED** [review]
Looks good, just some small changes.

**Comment by Zullybur** [comment]
Great!

**Comment by Zullybur** [comment]
Great!

**Review by mattcasiro — APPROVED** [review]
